From: Ben Hutchings <ben@decadent.org.uk
Date: Sat, 08 Mar 2014 19:53:37 +0000
Subject: net: Avoid ABI change in 3.2.55
Forwarded: not-needed

CONFIG_NET_DMA has been disabled as it can lead to data loss and
isn't going to be fixed.

To avoid an ABI change, keep defining the various structure members
that support it.


--- a/include/linux/skbuff.h
+++ b/include/linux/skbuff.h
@@ -446,8 +446,8 @@ struct sk_buff {
 
 	/* 0/13 bit hole */
 
-#ifdef CONFIG_NET_DMA
-	dma_cookie_t		dma_cookie;
+#ifdef CONFIG_NET_DMA_DUMMY
+	dma_cookie_t		dma_cookie; /* now unused */
 #endif
 #ifdef CONFIG_NETWORK_SECMARK
 	__u32			secmark;
--- a/include/linux/tcp.h
+++ b/include/linux/tcp.h
@@ -325,8 +325,8 @@ struct tcp_sock {
 		struct iovec		*iov;
 		int			memory;
 		int			len;
-#ifdef CONFIG_NET_DMA
-		/* members for async copy */
+#ifdef CONFIG_NET_DMA_DUMMY
+		/* members for async copy, now unused */
 		struct dma_chan		*dma_chan;
 		int			wakeup;
 		struct dma_pinned_list	*pinned_list;
--- a/include/net/sock.h
+++ b/include/net/sock.h
@@ -284,8 +284,8 @@ struct sock {
 	struct sk_filter __rcu	*sk_filter;
 	struct socket_wq __rcu	*sk_wq;
 
-#ifdef CONFIG_NET_DMA
-	struct sk_buff_head	sk_async_wait_queue;
+#ifdef CONFIG_NET_DMA_DUMMY
+	struct sk_buff_head	sk_async_wait_queue; /* now unused */
 #endif
 
 #ifdef CONFIG_XFRM
--- a/drivers/dma/Kconfig
+++ b/drivers/dma/Kconfig
@@ -262,6 +262,11 @@ config NET_DMA
 	  Say Y here if you enabled INTEL_IOATDMA or FSL_DMA, otherwise
 	  say N.
 
+config NET_DMA_DUMMY
+	bool
+	depends on DMA_ENGINE && NET
+	default (INTEL_IOATDMA || FSL_DMA)
+
 config ASYNC_TX_DMA
 	bool "Async_tx: Offload support for the async_tx api"
 	depends on DMA_ENGINE
